src/main/java/tc/oc/pgm/discord/DiscordClient.java

```diff
@@ -0,0 +1,37 @@
+package tc.oc.pgm.discord;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import javax.security.auth.login.LoginException;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.JDABuilder;
+import net.dv8tion.jda.api.entities.User;
+import tc.oc.pgm.Config;
+
+public class DiscordClient {
+
+  private static JDA jda;
+  public static final Cache<UUID, String> TOKENS_CACHE =
+      CacheBuilder.newBuilder().expireAfterWrite(15, TimeUnit.MINUTES).build();
+
+  public static void initialize() throws InterruptedException, LoginException {
+    jda = new JDABuilder(Config.Discord.token()).build();
+    jda.awaitReady();
```

Comment:
This will mane the server startup slower, considerhavong ready-state handled async, and just not enabling doscord integration features until it reaches a ready state

Reply:
Consider doing whatever needs to be done once the client is ready in the `OnReady` event.